src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/ApplyChangesToWorkspaceContext.cs

```diff
@@ -150,8 +150,15 @@ private void ProcessOptions(IProjectRuleSnapshot snapshot)
         {
             Assumes.NotNull(_context);
 
+            // The order of options matters and we should use warnaserror- and then warnaserror+
+            // Depending which comes first, the compiler produces different diagnostics.
+            // We just need to move warnaserror+ to the end.
+            var warnaserrorPlus = snapshot.Items.Where(a => a.Key.Contains("/warnaserror+")).FirstOrDefault();
+            ImmutableArray<string> commandLineArguments = warnaserrorPlus.Key is null ? snapshot.Items.Keys.ToImmutableArray() : 
+                snapshot.Items.Remove(warnaserrorPlus.Key).Keys.ToArray().Append(warnaserrorPlus.Key).ToImmutableArray();
+ 
             // We just pass all options to Roslyn
-            _context.SetOptions(snapshot.Items.Keys.ToImmutableArray());
+            _context.SetOptions(commandLineArguments);
```

Comment:
nit: Can be simplified. `ToArray` resolves the query, `Append` creates a new query, and `ToImmutableArray` resolves that new query. So, it is inefficient for performance and memory.
```suggestion
            IEnumerable<string> commandLineArguments = warnaserrorPlus.Key is null ? snapshot.Items.Keys :
                snapshot.Items.Remove(warnaserrorPlus.Key).Keys.Append(warnaserrorPlus.Key)

            // We just pass all options to Roslyn
            _context.SetOptions(commandLineArguments.ToImmutableArray());
```
In this way, the call to `SetOptions` will resolve the query, independent of the ternary's result.